### PR TITLE
feat: added near-workspace-js typedoc

### PR DESCRIPTION
**Pull request by matiasbenary**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +30,8 @@
     "husky": "^7.0.1",
     "lerna": "^5.2.0",
     "ts-node": "^10.1.0",
-    "typescript": "^4.3.5",
+    "typedoc": "^0.25.4",
+    "typescript": "^4.6.X",
```
[the remark this answers is not in the document]
Comment (gagdiez) [reply package.json:34]: we need to revert the typescript change and use `typedoc v0.24.8`

File: packages/js/__tests__/jsonrpc.ava.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import process from 'process';
 
 import test from 'ava';
 
-import {JsonRpcProvider} from '..';
+import {JsonRpcProvider} from '../dist/index.js';
```
Comment (gagdiez) [inline packages/js/__tests__/jsonrpc.ava.js:5]: @ailisp we needed to make this small change in order to make it compatible with the lint

**Review by gagdiez — APPROVED** [review]

**Comment by gagdiez** [comment]
@ailisp this adds automatic docs to the repo, it looks good to me, please merge if you agree with it as well.

**Comment by ailisp** [comment]
Looks great!

**Pull request merged**